quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py

```diff
@@ -0,0 +1,699 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Demo overview
+# MAGIC This notebook shows you how to use Mosaic AI to evaluate and improve the quality/cost/latency of a tool-calling Agent, deploying the resulting Agent to a web-based chat UI.
+# MAGIC
+# MAGIC Using Mosiac AI [Agent Evaluation](https://docs.databricks.com/en/generative-ai/agent-evaluation/index.html), [Agent Framework](https://docs.databricks.com/en/generative-ai/agent-framework/build-genai-apps.html), [MLflow](https://docs.databricks.com/en/generative-ai/agent-framework/log-agent.html) and [Model Serving](https://docs.databricks.com/en/generative-ai/agent-framework/deploy-agent.html), we will:
+# MAGIC 1. Generate synthetic evaluation data from a document corpus
+# MAGIC 2. Create a tool-calling Agent with a Retriever tool
+# MAGIC 3. Evaluate the Agent's quality/cost/latency across several foundational models
+# MAGIC 4. Deploy the Agent to a web-based chat app
+# MAGIC
+# MAGIC Requirements: 
+# MAGIC * Use a Serverless or MLR/DBR 14.3+ compute cluster
+# MAGIC * Databricks Serverless & Unity Catalog enabled
+# MAGIC * CREATE MODEL access to a Unity Catalog schema
+# MAGIC * Permission to create Model Serving endpoints
+# MAGIC
+# MAGIC **How to use this notebook**:
+# MAGIC 1. Update the following cell to configure the UC location where the agent model will be register to
+# MAGIC
+# MAGIC  ✅✏️ CONFIGURE THE UC MODEL NAME HERE
+# MAGIC
+# MAGIC 2. Click Run all
+# MAGIC
+# MAGIC ![alt text](https://github.com/databricks/genai-cookbook/blob/demo_gifs/quick_start_demo/demo_gifs/demo_overview.gif?raw=true "Title")
+# MAGIC
+# MAGIC See this [YouTube channel](https://www.youtube.com/@EricPeter-q6o) if you'd like to watch videos that go deeper into the capabilities.
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # 0/ Setup
+
+# COMMAND ----------
+
+# MAGIC %pip install -U -qqqq databricks-agents mlflow-skinny databricks-sdk[openai]
+# MAGIC dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# MAGIC %load_ext autoreload
+# MAGIC %autoreload 2
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## 0.1/ Configure user name to avoid conflicts with other users running the demo
+
+# COMMAND ----------
+
+import mlflow
+from databricks.sdk import WorkspaceClient
+
+# Get current user's name & email to ensure each user doesn't over-write other user's outputs
+w = WorkspaceClient()
+user_email = w.current_user.me().user_name
+user_name = user_email.split("@")[0].replace(".", "_")
+
+print(f"User name: {user_name}")
+
+# COMMAND ----------
+
+########################
+# #### ✅✏️ CONFIGURE THE UC MODEL NAME HERE
+########################
+#UC_MODEL_NAME = f"<ENTER_YOUR_CATALOG_NAME>.<ENTER_YOUR_SCHEMA_NAME>.db_docs__{user_name}"
+UC_MODEL_NAME = f"yz_agent_demo.playground.db_docs__{user_name}"
```

Comment:
rename this to 'catalog.schema.my_demo_agent' to improve the clarity

Reply:
agree, updated